tests/lib/spread/rules/main.yaml

```diff
@@ -0,0 +1,23 @@
+rules:
+  tests:
+    from:
+      - tests/main/.*
+      - tests/core/.*
+      - tests/completion/.*
+      - tests/cross/.*
+      - tests/regression/.*
+      - tests/smoke/.
+      - tests/unit/.*
+      - tests/upgrade/.*
+    to: [$SELF]
+
+  unit:
+    from: [.*_test.go]
```

Comment:
Sorry, I haven't reviewed the filter tool, how does the logic for matching the rules work? Is the resulting set of tests a union of the results from each ruleset? i.e. such that a PR modifying foo_test.go and tests/main/foo will trigger both `unit` and `tests` rulesets?

Reply:
right, in that case it executes both

Reply:
the filter get all the changes and calculates the tests to be executed for each change and makes an union for all the tests to execute